onsp_co2_employee_commuting/models/co2_commuting.py

```diff
@@ -0,0 +1,18 @@
+
+from odoo import models, fields, api
+
+
+
+class Co2Commuting(models.Model):
+    _name = "co2.commuting"
+
+
+    carbon_factor_id = fields.Many2one('carbon.factor', domain="[('id', 'in', allowed_carbon_factor_ids),('carbon_value', '!=', False)]", string="Transport mean")
+    allowed_carbon_factor_ids = fields.Many2many('carbon.factor', related="hr_employee_id.company_id.employee_commuting_carbon_factor.descendant_ids", domain="[('carbon_uom_id', 'ilike', 'km')]")
+    distance_km = fields.Integer(string="Average weekly distance in kilometers")
+    hr_employee_id = fields.Many2one('hr.employee', string="Employee")
+
+
+    def get_carbon_value(self):
+        for commuting in self:
+            return commuting.carbon_factor_id.carbon_value * commuting.distance_km
```

Comment:
I don't understand this method:
- Either this is a multi method that is probably bugged (it will return a value for the first record only)
- Or it's a single method and you don't need to loop on commutings + it lacks `self.ensure_one()`

Also, why not a computed field? Something like:

```suggestion
    carbon_value = fields.Float(compute="_compute_carbon_value", digits="Carbon value")
    
    @api.depends('carbon_factor_id.carbon_value', 'distance_km')
    def _compute_carbon_value(self):
        for commuting in self:
            commuting.carbon_value = commuting.carbon_factor_id.carbon_value * commuting.distance_km
```